docs/source/transactions/life_of_a_transaction.rst

```diff
@@ -88,9 +88,12 @@ The ordering of these steps is important:
    in between our start and commit, and someone else deleted the value after our commit. If we pass our lock check, but
    then lose our locks and have a very long GC, Thorough Sweep might clear all evidence of the conflict, meaning that
    we miss a read-write conflict. This would be safe for conservative sweep because of the deletion sentinel.
-8. The lock check may be run in parallel with pre-commit condition checks, though it must strictly be run before writing
+8. We need to check that the pre-commit conditions, if any, still hold. Given that these checks are user defined and
+   could involve transactional reads to the database, they require the same protections as the read-write conflict
+   checks in the prior step. An alternative implementation of those checks could be done as a pre-commit condition, for
+   example, so it is important that we handle this before the final lock check.
+9. The lock check may be run in parallel with pre-commit condition checks, though it must strictly be run before writing
```

Comment:
I guess this was moved down directly, but based on the logic in 8 I don't think we can say that it can be run in parallel any more. It's probably fine for this to just say "The lock check must strictly be run before writing ..."

Reply:
Yes, good catch, I just changed the order without reading :(